plugins/modules/notification/discord.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Christian Wollinger <cwollinger@web.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: discord
+short_description: Send discord messages
+description:
+  - Send a messages to a discord channel via the discord webhook API.
+author: Christian Wollinger (@cwollinger)
+notes:
+  - "Find the API documentation for Discord API here: U(https://discord.com/developers/docs/resources/webhook#execute-webhook)."
+options:
+  webhook_id:
+    description:
+      - "Format from discord webhook URL: C(/webhooks/{webhook.id}/{webhook.token})."
+    required: yes
+    type: str
+  webhook_token:
+    description:
+      - "Format from discord webhook URL: C(/webhooks/{webhook.id}/{webhook.token})."
+    required: yes
+    type: str
+  content:
+    description:
+      - Content of the message to the discord channel.
+    type: str
+  username:
+    description:
+      - Overrides the default username of the webhook.
+    type: str
+  avatar_url:
+    description:
+      - Overrides the default avatar of the webhook.
+    type: str
+  tts:
+    description:
+      - Set this to C(true) if this is a TTS (Test to Speech) message.
+    type: bool
+    default: false
+  embeds:
+    description:
+      - Send messages as Embeds to the discord channel.
+      - Embeds can have a colored border, embedded images, text fields and more.
+    type: list
+    elements: str
```

Comment:
```suggestion
    elements: dict
```
The elements clearly are dictionaries, not strings, in your examples.

If this isn't a complete free-form dict, but there are only certain allowed fields, you should make documentation and the accepted schema more precise (`suboptions:` in the documentation, `options=dict(...)` in the argument spec).

Reply:
Hi @felixfontein,

yes, the embeds dict is not free-form. There are many options available: https://discord.com/developers/docs/resources/channel#embed-object-embed-structure

Should I document all of these in the spec?

Regards
CWollinger

Reply:
@CWollinger whoa, that's quite a lot. I would suggest that being in a future PR, so we do not hold this one back - it will take a while to chew through all that multi-level data structure. @felixfontein @aminvakil what do you think?

Reply:
My only concern is not to depend this module on upstream API heavily, so that it will break if they deprecate stuff and after someone fixes module, old users who have not updated to latest module cannot use options only because old module does not let them do it, documenting options is good though.

Reply:
@russoz adding that later would be a breaking change. So either add it now, or never.

Reply:
I would suggest to reference to the Discord API Docs and add another example.